src/routes/transactions/mappers/common/native-staking.mapper.ts

```diff
@@ -248,6 +319,14 @@ export class NativeStakingMapper {
 
     return 0;
   }
+
+  private getPublicKeysFromDataDecoded(data: DataDecoded): `0x${string}`[] {
+    const publicKeys =
+      data.parameters?.filter(
+        (parameter) => parameter.name === '_publicKeys',
+      ) ?? [];
+    return publicKeys.map((publicKey) => publicKey.value as `0x${string}`);
```

Comment:
Nit: we can use the `toHex` helper to avoid casting here.

Reply:
Thanks, but casting is still needed if we change that as `publicKey.value` is of type `Required<unknown>`, right? Also, I've noticed the `toHex` function interprets the input parameter as an UTF8 string instead of an hex?

```
> viem.toHex('0x0001')
'0x307830303031'
```

Reply:
Very true. You'd need to do some other mapping/filtering beforehand, e.g.

```
return publicKeys
      .map((publicKey) => publicKey.value)
      .filter((publicKey) => typeof publicKey === 'string')
      .map((publicKey) => toHex(publicKey))
```

I'm happy to keep it as is.

Do you think we should fallback to manual decoding if the no `dataDecoded` was returned?

Reply:
Sounds good to me.